File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        
        int start = 0, end = height.size()-1;
        int maxArea = 0;
        while(start < end)
        {
            int area = (end - start)*min(height[start], height[end]);
            maxArea = max(area, maxArea);
            if(height[start] < height[end]){
                start++;
            }
            else{
                end--;
            }
        }
        return maxArea;
    }
};

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int n = height.length;
        int start = 0, end = n-1;
        int ans = 0;
        while(start < end)
        {
            //System.out.println(start + " " + end);
            int val = Math.min(height[start], height[end])*(end - start);
            ans = Math.max(val, ans);
            //System.out.println(val);
            if(height[start] < height[end])
                start++;
            else
                end--;
        }
        return ans;
    }
}